preppal-fe/src/pages/recipe-calendar.tsx

```diff
@@ -0,0 +1,236 @@
+      }
+    }
+    else {
+      // set to default
+      setRecipeID("");
+      setRecipeTitle("");
+      setShoppingList("");
+    }
+  }
+
+  // runs when a recipe is selected for a day in the calendar
+  const updateRecipeAndShoppingList = (event: any) => {
+    if (event && event?.target && ownRecipes) {
+      const thisID = (event.target as HTMLInputElement).value
+      // using thisID for a recipe, get the recipe index and set global variables
+      const index = ownRecipes.findIndex((recipe: any) => recipe._id === thisID);
+      if (index > -1) {
+        // if valid ID in ownRecipes, set global variables
+        setRecipeID(ownRecipes[index]._id);
+        setRecipeTitle(ownRecipes[index].title);
+        const shoppingList = ownRecipes[index].ingredients.join(", ");
+        setShoppingList(shoppingList);
+      }
+      else {
+        // set to default
+        setRecipeID("");
+        setRecipeTitle("");
+        setShoppingList("");
+      }
+    }
+  }
+
+  // when update calendar day clicked, submit current recipe and date to api to update backend
+  async function submitDay() {
+    try {
+      const currDateString = currDate?.toLocaleString().split(",")[0];
+      if (currDateString !== undefined) {
+        const req = {
+          method: "POST",
+          headers: {
+            'Content-Type': 'application/json'
+          },
+
+          body: JSON.stringify({
+            "username": username,
+            "dateIs": currDateString,
+            "recipeOfTheDayID": recipeID,
+            "recipeOfTheDayTitle": recipeTitle,
+            "recipeOfTheDayIngredients": shoppingList
+          })
+        };
+        const res = await fetch("http://localhost:9001/api/calendar/updateCalendar/", req).then(res => res.json());
+        setCalendarObject(res);
+        // returns new calendar
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+
+  return (
+    <><NavBar></NavBar>
+      <Row className='p-3 d-flex'>
+        <div className="w-100" style={{ maxWidth: '2000px' }}>
+          <Card className='p-4 d-flex' style={{ backgroundColor: "#F2E8DC" }}>
+            <Row>
+              <Col>
+                <div className='d-flex p-4 justify-content-center' style={{ maxWidth: '500px' }}>
+                  <Calendar className={'justify-content-center'} onChange={updateDayAndRecipe} value={currDate} />
+                </div>
+              </Col>
+              <Col className='d-flex p-4 justify-content-center align-items-center'>
+                <div>
+                  <Stack className='d-flex p-4'>
+                    <label>{currDate?.toLocaleString().split(",")[0]}</label>
+                    <label className='py-4'>Recipe of the day: {recipeTitle}</label>
+                    <div>
+                      <Button className="mx-auto" variant="primary" onClick={submitDay} title="SubmitUpdate" size="sm" style={{ backgroundColor: "#401E01" }}>
+                        Update calendar day
```

Comment:
This button was a bit confusing to me. I think `Save/update recipe to this day` might be better?